youtube_dl/extractor/funimation.py

```diff
@@ -0,0 +1,110 @@
+            'email_field': username,
+            'password_field': password,
+        })
+        login_request = sanitized_Request(login_url, data)
+        login_request.add_header('Content-Type', 'application/x-www-form-urlencoded')
+        self._download_webpage(login_request, login_url)
+
+    def _real_initialize(self):
+        self._login()
+
+class FunimationIE(FunimationBaseIE):
+    _VALID_URL = r'(?P<address>https?://(?:www\.)?funimation\.com/shows/.*/videos/official/.+)'
+    _USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/601.2.7 (KHTML, like Gecko) Version/9.0.1 Safari/601.2.7'
+
+    _TEST = {
+        'url': 'http://www.funimation.com/shows/noragami/videos/official/a-housecat-a-stray-god-and-a-tail?watch=sub',
+        'info_dict': {
+            'id': 'NRGENG0001',
+            'title': 'Noragami - A Housecat, a Stray God, and a Tail',
+            'ext': 'mp4',
+        },
+        'params': {
+            'username': 'foo@bar.com',
+            'password': '(snip)',
+        },
+        'skip': 'requires actual password',
+    }
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+
+        request = sanitized_Request(url)
+        request.add_header('User-Agent', self._USER_AGENT)
+
+        video_address = mobj.group('address')
+        webpage = self._download_webpage(request, video_address)
+
+        self.report_extraction(video_address)
+
+        video_base = self._search_regex(r'"sdUrl":"(.+?)(?:-1500K.mp4)"', webpage, 'video_base')
+        video_id = self._search_regex(r'"FUNImationID":"(.+?)"', webpage, 'video_id')
+        video_auth = self._search_regex(r'"authToken":"(.+?)"', webpage, 'video_auth')
+
+        video_base = video_base.replace('\\', '')
+
+        formats = [{
+            'format_id': 'lowest',
+            'quality': 1,
+            'vbr': 1000,
+            'resolution': '480p',
+            'url': video_base + '-1000K.mp4' + video_auth,
+        }, {
+            'format_id': 'lower',
+            'quality': 2,
+            'vbr': 1500,
+            'resolution': '480p',
+            'url': video_base + '-1500K.mp4' + video_auth,
+        }, {
+            'format_id': 'low',
+            'quality': 3,
+            'vbr': 2000,
+            'resolution': '480p',
+            'url': video_base + '-2000K.mp4' + video_auth,
+        }, {
+            'format_id': 'medium',
+            'quality': 4,
+            'vbr': 2500,
+            'resolution': '720p',
+            'url': video_base + '-2500K.mp4' + video_auth,
+        }, {
+            'format_id': 'high',
+            'quality': 5,
+            'vbr': 3500,
+            'resolution': '720p',
+            'url': video_base + '-3500K.mp4' + video_auth,
+        }, {
+            'format_id': 'highest',
+            'quality': 6,
+            'vbr': 4000,
+            'resolution': '1080p',
+            'url': video_base + '-4000K.mp4' + video_auth,
+        }]
```

Comment:
Actually, it seems I was wrong about Hulu, when you login, even 10 year old videos becomes Funimation hosted. I checked 50 or so from different seasons going back 10 years. 

Only the video without login are Hulu hosted, it seems. I already have an error for not providing login information.

About the formats, all formats have different urls. I tried what you wrote but I have problems with url. I did it like this:

        formats_map = [
            ('lowest', 480, 1000),
            ('lower', 480, 1500),
            ('low', 480, 2000),
            ('medium', 720, 2500),
            ('high', 720, 3500),
            ('highest', 1080, 4000),
        ]
        formats = [{
            'format_id': format_id,
            'vbr': video_bitrate,
            'resolution': height + 'p',
            'url': video_base + '-' + video_bitrate + '.mp4',
        } for format_id, height, video_bitrate in formats_map]
```
What I am getting wrong? It says *TypeError: unsupported operand type(s) for +: 'int' and 'unicode'*.